flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistry.java

```diff
@@ -216,17 +244,20 @@ public ScopeFormats getScopeFormats() {
 	 * @param metricName  the name of the metric
 	 * @param group       the group that contains the metric
 	 */
-	public void register(Metric metric, String metricName, MetricGroup group) {
+	public void register(Metric metric, String metricName, AbstractMetricGroup group) {
 		try {
 			if (reporters != null) {
-				for (MetricReporter reporter : reporters) {
+				for (int i = 0; i < reporters.size(); i++) {
+					MetricReporter reporter = reporters.get(i);
 					if (reporter != null) {
-						reporter.notifyOfAddedMetric(metric, metricName, group);
+						FrontMetricGroup front = new FrontMetricGroup(i);
+						front.setReference(group);
```

Comment:
If we need use `FrontMetricGroup` also for `unregister()`, do I need return list of `FrontMetricGroup` ? and we will avoid for `register()` 
>we will now create a new object every time we add a new metric.

in this case